workflows/cmgg-preprocessing.nf

```diff
@@ -214,125 +231,136 @@ workflow CMGGPREPROCESSING {
+    | branch { meta, reads ->
+        supported: meta.genome
         other: true
     }
-    ch_trimmed_reads.human.dump(tag: "MAIN: human reads",pretty: true)
-    ch_trimmed_reads.other.dump(tag: "MAIN: other reads",pretty: true)
+    | set { ch_trimmed_reads }
 
-    /*
-    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-    // STEP: FASTQ TO UNALIGNED CRAM CONVERSION
-    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-    */
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// STEP: FASTQ TO UNALIGNED CRAM CONVERSION
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
 
-    FASTQ_TO_UCRAM(
-        ch_trimmed_reads.other,
-        ch_fasta_fai
-    )
+    FASTQ_TO_UCRAM(ch_trimmed_reads.other)
     ch_versions = ch_versions.mix(FASTQ_TO_UCRAM.out.versions)
 
-    /*
-    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-    // STEP: FASTQ TO ALIGNED CRAM CONVERSION
-    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-    */
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// STEP: FASTQ TO ALIGNED CRAM CONVERSION
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
 
     FASTQ_TO_CRAM(
-        ch_trimmed_reads.human,
-        ch_fasta_fai,
+        ch_trimmed_reads.supported,
         aligner,
-        ch_aligner_index,
         markdup
     )
+
     ch_multiqc_files = ch_multiqc_files.mix(FASTQ_TO_CRAM.out.multiqc_files)
     ch_versions = ch_versions.mix(FASTQ_TO_CRAM.out.versions)
 
-    /*
-    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-    // COVERAGE ANALYSIS
-    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-    */
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// STEP: COVERAGE ANALYSIS
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
 
     // Generate coverage metrics and beds for each sample
-    // COVERAGE([meta,bam, bai], [meta2, fasta, fai], target)
-    ch_cram_crai_branch = FASTQ_TO_CRAM.out.cram_crai
-        .combine(ch_target_regions)
-        .branch {
-            bed: it.size() == 4
-                return it
-            nobed: it.size() == 3
-                return it + [[]]
-            }
-    ch_cram_crai_target = ch_cram_crai_branch.bed
-        .mix(ch_cram_crai_branch.nobed)
-        .dump(tag: "MAIN: cram_crai_target",pretty: true)
+    FASTQ_TO_CRAM.out.cram_crai
+    | map { meta, cram, crai ->
+        if (meta.roi) {
+            return [
+                meta,
+                cram,
+                crai,
+                file(meta.roi, checkIfExists:true)
```

Comment:
This should already be a file object if you correctly configured the nf-validation schema, basically it removes the need for these kinds of if statements by directly casting all inputs to the correct type or to `null` or `[]` if they are missing